yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -1444,4 +1444,19 @@ object Client extends Logging {
     uri.startsWith(s"$LOCAL_SCHEME:")
   }
 
+  /**
+   *  Returns the app staging dir.
+   */
+  private def getAppStagingDirPath(
+      conf: SparkConf,
+      fs: FileSystem,
+      appStagingDir: String): Path = {
+    val stagingRootDir = conf.get(STAGING_DIR).orNull
```

Comment:
`conf.get(STAGING_DIR).orElse(fs.getHomeDirectory)` gives type mismatch compilation error since the `fs.getHomeDirectory` return type is Path and it expects to be String as per `conf.get(STAGING_DIR)`.